src/cache/cache_las.c

```diff
@@ -682,10 +683,28 @@ __wt_las_insert_updates(WT_CURSOR *cursor, WT_BTREE *btree, WT_PAGE *page, WT_MU
 err:
     /* Resolve the transaction. */
     if (local_txn) {
-        if (ret == 0)
+        if (ret == 0) {
             ret = __wt_txn_commit(session, NULL);
-        else
+            txn_rollbacked = ret != 0;
+        } else {
             WT_TRET(__wt_txn_rollback(session, NULL));
+            txn_rollbacked = true;
+        }
+
+        if (txn_rollbacked && multi->supd_entries > 0) {
+            /* We only need to clear the flag on the updates up to where the error occurs. */
+            err_pos = WT_MIN(multi->supd_entries - 1, i);
```

Comment:
Since you are re-using `i` from a previous loop it needs a more descriptive name - `i` is the name for a generic increment counter within a loop.

Reply:
fixed